src/main/java/com/vackosar/gitflowincrementalbuild/control/DifferentFiles.java

```diff
@@ -9,144 +12,196 @@
 import org.eclipse.jgit.revwalk.RevCommit;
 import org.eclipse.jgit.revwalk.RevWalk;
 import org.eclipse.jgit.revwalk.filter.RevFilter;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
 import org.eclipse.jgit.transport.RefSpec;
 import org.eclipse.jgit.treewalk.TreeWalk;
 import org.eclipse.jgit.treewalk.filter.TreeFilter;
 
 import javax.inject.Inject;
+import javax.inject.Named;
 import javax.inject.Singleton;
+
+import java.io.File;
 import java.io.IOException;
 import java.nio.file.Path;
 import java.util.HashSet;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 @Singleton
+@Named("gib.differentFiles")
 public class DifferentFiles {
 
+    public static final String UNSUPPORTED_WORKTREE = "JGit unsupported separate worktree checkout detected from current git dir path: ";
+
     private static final String HEAD = "HEAD";
     private static final String REFS_REMOTES = "refs/remotes/";
     private static final String REFS_HEADS = "refs/heads/";
-    @Inject private Git git;
-    @Inject private Configuration configuration;
+
     @Inject private Logger logger;
+    @Inject private MavenSession mavenSession;
 
-    public Set<Path> get() throws GitAPIException, IOException {
+    public Set<Path> get(Configuration configuration) throws GitAPIException, IOException {
         Set<Path> paths = new HashSet<>();
-        fetch();
-        checkout();
-        final Path workTree = git.getRepository().getWorkTree().toPath().normalize().toAbsolutePath();
-        if (!configuration.disableBranchComparison) {
-            paths.addAll(getBranchDiff(workTree));
-        }
-        if (configuration.uncommited || configuration.untracked) {
-            paths.addAll(getChangesFromStatus(workTree));
-        }
-        return paths;
-    }
 
-    private Set<Path> getBranchDiff(Path workTree) throws IOException {
-        RevCommit base = getBranchCommit(configuration.baseBranch);
-        final TreeWalk treeWalk = new TreeWalk(git.getRepository());
+        Git git = setupGit(configuration);
         try {
-            treeWalk.addTree(base.getTree());
-            treeWalk.addTree(resolveReference(base).getTree());
-            treeWalk.setFilter(TreeFilter.ANY_DIFF);
-            treeWalk.setRecursive(true);
-            return getDiff(treeWalk, workTree);
+            Worker worker = new Worker(git, configuration);
```

Comment:
Note: Instead of pasing `configuration` to each and every private method, I decided to wrap everything in an inner `Worker` class so that the `configuration` member approch could be retained.